### PR TITLE
Switch distance computation to total distance

### PR DESCRIPTION
**Pull request by rcoh**
Switch distance computation to total distance. Adds new service + test

### CODE REVIEW

File: README.rdoc [hunk 1]
```diff
@@ -1,28 +1,11 @@
...
-
-* Services (job queues, cache servers, search engines, etc.)
-
-* Deployment instructions
-
-* ...
-
-
-Please feel free to use a different markup language if you do not plan to run
-<tt>rake doc:app</tt>.
```
Comment (alexneigher) [inline README.rdoc]: thanks for these updates @rcoh :)

**Comment by alexneigher** [comment]
@rcoh I think this is going to end up being a super great improvement. How would you feel about listing both the "as the crow flies" distance, as well as the total infinite waypoint distance in the list? I could see it something like this:
screen shot 2018-04-30 at 9 51 07 pm


For the mock I removed the timestamp, but I think having at least an hour/minute of the most recent trackpoint is pretty valuable...

Thoughts?

**Comment by rcoh** [comment]
+1 I think that's awesome

**Comment by alexneigher** [comment]
Just for my own curiosity I'm going to benchmark this against a list of tracklogs of ~25 waypoints, and compare it to the single distance calc. I am very sensitive to performance changes, since performance and usability were my original motivation for wanting to build this over xcfind.

If there is a significant decrease in page load time, I would be happy to convert this to an ajax call that fetches the flight metadata (including things like running/walking/flying/etc), that would run asynchronously.

**Comment by rcoh** [comment]
boom: 
selection_017

File: app/services/track_length_service.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'enumerator'
+class TrackLengthService
+  def initialize(track)
+    @track = track.sort_by(&:timestamp)
```
Comment (alexneigher) [inline app/services/track_length_service.rb]: i think `sort_by()` is a ruby method, where as `.order()` is going to tell active record to write sql: `order by`. I'd prefer to sql ordering over ruby sorting if possible

File: app/models/most_recent_flight.rb [hunk 3]
```diff
@@ -8,11 +8,11 @@ def has_waypoints?
 
   def flight_length
     return 0 if waypoints.length < 2
-    
-    ordered_waypoints = waypoints.order(:timestamp)
```
Comment (alexneigher) [inline app/models/most_recent_flight.rb]: here's the `order()`

**Comment by rcoh** [comment]
yeah, I switched it to sort_by so it's testable. With ~100 points in the
extreme case it's not going to make a difference.

**Comment by alexneigher** [comment]
@rcoh However 100points x 50 users is going to invoke quite a bit of sorting. What happens w/ the testing when you try to keep the query in active record using `order`? Is it simply an association issue? where `waypoints` returns `[]`? if that is the case, it's pretty easy to mock up a `most_recent_flight` with many associated `waypoints`

**Comment by rcoh** [comment]
Did a quick test. Sorting 1000 tracks with 100 points per track takes `0.012` seconds

**Comment by alexneigher** [comment]
@rcoh great thanks for that! 🚢- it

**Pull request merged**